### PR TITLE
WIP: Prototypes to improve our current navigation design.

### PR DESCRIPTION
**Pull request by HarshulSharma000**
This one is a pull request related to Nested navigators isolated discussion in mobile kindly refer to that first. Will be posting a description of this PR in there.

### CODE REVIEW

**Comment by ebouJ** [comment]
You can test your changes locally before pushing it by running "npm run test:lint"

File: src/main/MainScreenWithTabs.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ export default class MainScreenWithTabs extends PureComponent<{}> {
   static contextTypes = {
     styles: () => null,
   };
+  static router = MainTabs.router; //Try to comment out this line and check navigation state Tab navigator routes are registered with this
```
Comment (HarshulSharma000) [inline src/main/MainScreenWithTabs.js]: Here we give a router to MainScreenWithTabs(the one which MainTabs.router is owning) by doing this whenever parent navigator(AppNavigator) looks in MainScreenWithTabs it finds a router corresponding to the screen. Thus it gets to know there are children routes to MainScreenWithTabs

**Comment by borisyankov** [comment]
React Navigation did release a release candidate of 2.0
There might be a final version within a week.

**Comment by HarshulSharma000** [comment]
@borisyankov  Cool that's good. But still they do have some bugs left that are planned to be resolved by v2 https://github.com/react-navigation/react-navigation/issues/3685

**Comment by borisyankov** [comment]
Once 2.0 is released we should implement whatever becomes the `orthodox` approach.

**Pull request closed without merging**